compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1174,6 +1174,18 @@ private boolean isErrorTypeAssignable(BErrorType source, BErrorType target, Set<
                 && target.typeIdSet.isAssignableFrom(source.typeIdSet);
     }
 
+    private boolean isErrorTypeAssignableWithNoTypeIds(BErrorType source, BErrorType target,
+                                                       Set<TypePair> unresolvedTypes) {
+        if (target == symTable.errorType) {
+            return true;
+        }
+        TypePair pair = new TypePair(source, target);
+        if (!unresolvedTypes.add(pair)) {
+            return true;
```

Comment:
Can we try to add a test for this

Reply:
I have tried a couple of scenarios but still, I was unable to find a test case for this. So I will remove this, If we get an issue later we can consider adding this. WDYT?